### DISCUSSION TITLE
Specify Front or Back Camera - MediaPicker CapturePhotoAsync()

### DISCUSSION THREAD

**Discussion by rossatkinson265** [dotnet/maui#9520] [Ideas]
When opening the device's camera I would like to specify that the camera should open on the front camera.

I am using MediaPicker.Default.CapturePhotoAsync() to open the device's camera and capture a photo. 

I am developing a tablet application and the behaviour I have observed on test devices is as follows:

**Behaviour on Android Samsung tablet:**
Camera remembers previous settings, therefore if the user changes to front camera on first open, the subsequent open will also be front camera. This is good because the user only has to switch to front camera once.

**Behaviour on iOS iPad:**
Camera always defaults to rear camera, regardless of previous camera settings. This is frustrating because the user has to switch to front camera each time they clock in.

**Suggestion:**
I notice that the CapturePhotoAsync method can accept MediaPickerOptions as a parameter. If feasible, it would be nice to be able to specify which camera the device should open with, as part of the MediaPickerOptions parameter, or potentially overload the method and add a CameraOptions parameter, which would enable us to specify front or rear camera.

**Comment by rossatkinson265** [comment]
I have written my own custom implementation of this for iOS and Android based on the source code.

_MediaPicker.shared.cs_
Added new 'CameraType' enum to specify front or rear camera.
Included a new 'CameraType' property in the MediaPickerOptions class.
![image](https://user-images.githubusercontent.com/53495202/186852747-9d08e417-f43d-48e7-877f-39118074db72.png)

_MediaPicker.android.cs_
In the 'CaptureAsync' method, if front camera has been specified add extra intent to the capturePhotoIntent.
![image](https://user-images.githubusercontent.com/53495202/186853198-d5f8f56a-e8e1-49fa-8b56-f86e2312f969.png)
This does not seem to work consistently, however, as described in the original idea, Android maintains the previous camera settings so it is not as much of an issue. If there was a way to specify which camera to use on Android without using intents I believe that would be preferable.

_MediaPicker.ios.cs_
In the 'PhotoAsync' method, if we received a value for CameraType in the MediaPickerOptions, use this to set the CameraDevice.
![image](https://user-images.githubusercontent.com/53495202/186853683-07962e03-02e4-4c97-b772-119eac2aa0b5.png)
This works perfectly and opens the front camera if CameraType.Front was specified.

It would be cool if somebody could review this suggestion and the example code shown here and, if suitable, include it in a future release of the MAUI Essentials framework.

Thanks

Ross

**Reply by ricavir11** [reply]
Can you please share your code @rossatkinson265  ?

**Reply by rossatkinson265** [reply]
Hi - I'm no longer working at the company where I implemented this so I don't have access to the code but the screen clips in my comment show the changes I made to allow you to specify front or back camera. It wasn't a particularly maintainable solution because you have to overwrite the MAUI package with your altered 'custom' package every time you want to update to the latest version of MAUI Essentials - the code snips were more of a demonstration of how the framework could be changed to allow us to specify front or back camera on launch.

It's a shame that the MAUI team haven't picked this up as it seems like useful functionality for anyone wanting to use the camera in their application, but it's not looking likely that they will as this suggestion was originally made over 2 years ago! Looks like it's been raised in some other threads as well, https://github.com/dotnet/maui/issues/25032, but doesn't seem like they're going to prioritise it unfortunately.

**Reply by ricavir11** [reply]
Hi, Thank you for your feedback.
I created an issue yesterday that has already been closed this morning... one more request in the MAUI backlog.
I've just written a comment [here](https://github.com/dotnet/maui/issues/23596#issuecomment-2519799945) 
Hard days for old xamarin forms fellows
